reVX/turbine_flicker/turbine_flicker.py

```diff
@@ -0,0 +1,563 @@
+
+        flicker_excl[row_idx, col_idx] = 1
+
+        if out_layer:
+            logger.info('Saving flicker exclusions to {} as {}'
+                        .format(self._excl_h5, out_layer))
+            description = ("Pixels with value 1 will cause greater than {} "
+                           "hours of flicker on buildings in {}. Shadow "
+                           "flicker is computed using a {}m turbine."
+                           .format(flicker_threshold, self._bld_layer,
+                                   hub_height))
+            ExclusionsConverter._write_layer(self._excl_h5, out_layer,
+                                             profile, flicker_excl,
+                                             description=description)
+
+        return flicker_excl
+
+    @classmethod
+    def run(cls, excl_fpath, res_fpath, building_layer, hub_height,
+            tm_dset='techmap_wtk', building_threshold=0,
+            flicker_threshold=30, resolution=128,
+            max_workers=None, out_layer=None):
+        """
+        Exclude all pixels that will cause flicker exceeding the
+        "flicker_threshold" on any building in "building_layer". Buildings
+        are defined as pixels with >= the "building_threshold value in
+        "building_layer". Shadow flicker is computed at the supply curve point
+        resolution based on a turbine with "hub_height" (m) and applied to all
+        buildings within that supply curve point sub-array.
+
+        Parameters
+        ----------
+        excl_fpath : str
+            Filepath to exclusions h5 file. File must contain "building_layer"
+            and "tm_dset".
+        res_fpath : str
+            Filepath to wind resource .h5 file containing hourly wind
+            direction data
+        building_layer : str
+            Exclusion layer containing buildings from which turbine flicker
+            exclusions will be computed.
+        hub_height : int
+            Hub-height in meters to compute turbine shadow flicker for
+        tm_dset : str, optional
+            Dataset / layer name for wind toolkit techmap,
+            by default 'techmap_wtk'
+        building_threshold : float, optional
+            Threshold for exclusion layer values to identify pixels with
+            buildings, values are % of pixel containing a building,
+            by default 0
+        flicker_threshold : int, optional
+            Maximum number of allowable flicker hours, by default 30
+        resolution : int, optional
+            SC resolution, must be input in combination with gid,
+            by default 128
+        max_workers : None | int, optional
+            Number of workers to use, if 1 run in serial, if None use all
+            available cores, by default None
+        out_layer : str, optional
+            Layer to save exclusions under. Layer will be saved in
+            "excl_fpath", by default None
+
+        Returns
+        -------
+        flicker_excl : ndarray
```

Comment:
Is this an exclusion array where 1 is excluded or an inclusion array where 1 is included? Applicable to all instances of this docstring.

Reply:
Very good catch, we should really re-name everything to be inclusion mask. will change

Reply:
Inclusion masks are way better + more useful anyway. I hate the naming issues around this.

Reply:
still one place you should fix: https://github.com/NREL/reVX/blob/c4fe5f472c63effd0792153b54195780985372aa/reVX/turbine_flicker/turbine_flicker.py#L592

Reply:
doh